apps/lend/src/components/PageLoanManage/LoanRepay/index.tsx

```diff
@@ -381,18 +390,41 @@ const LoanRepay = ({
             tokenBalance={userBalances?.borrowed}
             debt={state?.debt ?? '0'}
             handleInpChange={(userBorrowed) => {
-              const isFullRepay = BN(userBorrowed).isEqualTo(state?.debt ?? '0')
-              updateFormValues({ userBorrowed, isFullRepay })
-            }}
-            handleMaxClick={() => {
-              if (userBalances && state) {
-                let isFullRepay = BN(userBalances.borrowed).isGreaterThanOrEqualTo(state?.debt ?? '0')
-                const max = isFullRepay ? state.debt : userBalances.borrowed
-                updateFormValues({ userBorrowed: max, isFullRepay })
+              if (expectedBorrowed) {
+                updateFormValues({ userBorrowed, isFullRepay: false })
               } else {
-                updateFormValues({ userBorrowed: '', isFullRepay: false })
+                let isFullRepay = false
+                if (state) {
+                  const { borrowed: stateBorrowed, debt: stateDebt } = state
+                  isFullRepay = BN(BN(stateBorrowed).plus(userBorrowed)).isGreaterThanOrEqualTo(stateDebt)
+                }
+                updateFormValues({ userBorrowed, isFullRepay })
               }
             }}
+            handleMaxClick={async () => {
+              let userBorrowed = ''
+              let isFullRepay = false
+
+              if (+userBalances.borrowed === 0) {
+                userBorrowed = ''
```

Comment:
I would make this part simpler, without the if-else states:
```suggestion
                updateFormValues({ userBorrowed: '', isFullRepay: false })
                return
```